airflow/utils/state.py

```diff
@@ -39,6 +39,7 @@ class TerminalTIState(str, Enum):
     FAILED = "failed"
     SKIPPED = "skipped"  # A user can raise a AirflowSkipException from a task & it will be marked as skipped
     REMOVED = "removed"
+    UP_FOR_RETRY = "up_for_retry"  # We do not need to do anything actionable for this state, hence it is a terminal state.
```

Comment:
"terminal state" means the task is done; up for retry does not really feel like a terminal state....  since it's going to be retried....

Reply:
this highlights an ambiguity / conflict.  the TI-_try_ is done, but the _TI_ is not